### DISCUSSION TITLE
[Camera Support]: Dahua TIOC HDW3849 - Siren stopped working when switched to Frigate

### DISCUSSION THREAD

**Discussion by ph0nebooth** [blakeblackshear/frigate#13510] [Camera Support] [answered]
### Describe the problem you are having

I am having an issue with the siren not firing via API or web interface on my Dahua TIOC HDW3849.

I believe this is because the cameras support two-way audio and the siren is blocked. See #6077 Adding #backchannel=0 to the mainstream solves the issue.

Is there any way I can have two-way talk enabled via the frigate-hass-card _and_ have backchannel=0 set? For example, maybe I can configure Frigate to deliver video via channel 0 (mainstream) and audio via channel 2 (substream 2)? The siren uses channel 0.

I trigger the sirens via the Dahua Home Assistant integration based on switch and/or automation.

### Version

0.14.1-f4f3cfa

### What browser(s) are you using?

Safari

### Frigate config file

```yaml
https://pastebin.com/in6XRyqx
```

### Relevant Frigate log output

```shell
N/A
```

### Relevant go2rtc log output

```shell
N/A
```

### FFprobe output from your camera

```shell
Input #0, rtsp, from 'rtsp://admin:redacted@192.168.178.201':
  Metadata:
    title           : Media Server
  Duration: N/A, start: 0.128000, bitrate: N/A
  Stream #0:0: Video: h264 (Main), yuvj420p(pc, bt470bg/bt470bg/bt709, progressive), 3840x2160, 15 fps, 15 tbr, 90k tbn
  Stream #0:1: Audio: aac (LC), 8000 Hz, mono, fltp
```

### Operating system

Other Linux

### Install method

Docker Compose

### Object Detector

Coral

### Network connection

Wired

### Camera make and model

Dahua TIOC HDW3849

### Screenshots of the Frigate UI's System metrics pages

<img width="1386" alt="Screenshot 2024-09-03 at 1 39 56 PM" src="https://github.com/user-attachments/assets/19c9e4da-9bd2-42c3-af86-c1cfa889424d">
<img width="1380" alt="Screenshot 2024-09-03 at 1 40 08 PM" src="https://github.com/user-attachments/assets/b0cc62d4-5add-4d89-8a31-7528b86438d0">

**Comment by NickM-27** [comment]
Add another steam to the list without backchannel=0

**Reply by ph0nebooth** [reply]
Thanks Nick. Like this?

    Terrace:
      - rtsp://cctv:redacted@192.168.178.202:554/cam/realmonitor?channel=1&subtype=0#backchannel=0
      - rtsp://cctv:redacted@192.168.178.202:554/cam/realmonitor?channel=1&subtype=0
      - ffmpeg:Terrace#audio=opus

I tried the above and two-way doesn't work. Incoming audio from the camera is super 'choppy' as well.

I've tried adding the substream as a second channel too. Incoming audio is fine this way, but two-way audio via the frigate-hass-card doesn't work.

**Reply by NickM-27** [reply]
this is how you do it, I have multiple cameras setup this way and it works as expected in the frigate UI, so it should work fine in the frigate card as well

**Reply by NickM-27** [reply]
you should change the order though

```
Terrace:
  - rtsp://cctv:redacted@192.168.178.202:554/cam/realmonitor?channel=1&subtype=0#backchannel=0
  - ffmpeg:Terrace#audio=opus
  - rtsp://cctv:redacted@192.168.178.202:554/cam/realmonitor?channel=1&subtype=0
```

**Reply by ph0nebooth** [reply]
I've changed it to that order and now incoming audio is fine but still no outgoing audio.

I reversed out all config changes and put it back to original. Rebooted Frigate and HA. Two-way isn't working like it was before (with the siren broken). 

I'm wondering if something else is at play here, unrelated to Frigate/Frigate card. The microphone doesn't even work in the go2rtc web interface.

**Reply by ph0nebooth** [reply]
Accessing the Frigate UI via SSL doesn't even display a mic option.

<img width="282" alt="Screenshot 2024-09-03 at 10 41 56 PM" src="https://github.com/user-attachments/assets/6abd2638-8a43-4bae-921b-e05b18eb5e24">

ffmpeg probe from the camera: https://pastebin.com/qGEW7xG7

**Reply by NickM-27** [reply]
you should use the `probe` button in the go2rtc UI

**Reply by ph0nebooth** [reply]
That's what I clicked to get the above dump :)

**Reply by NickM-27** [reply]
might need to restart the camera

**Reply by ph0nebooth** [reply]
Sorry, it got late last night so I didn't have a chance to reset the camera. I've reset the camera and it made no difference.

I had a look at the ffmpeg probe (I'm no expert) and saw the backchannel stream is using PCMA/8000. I changed the cameras audio codec from AAC to PCMA and now two-way works again. However, the siren still fails.

Below is the changed config. Everything else remains the same as my initial dump

    Terrace:
      - rtsp://cctv:redacted@192.168.178.202:554/cam/realmonitor?channel=1&subtype=0#backchannel=0
      - ffmpeg:Terrace#audio=opus
      - rtsp://cctv:redacted@192.168.178.202:554/cam/realmonitor?channel=1&subtype=0
      
 EDIT: disregard my reference trying PCMA. When you update the codec on the camera, the siren and two-way work for 5-10 seconds before locking up again.

ffmpeg stream info: https://pastebin.com/Vm7W5Hkg
ffmpeg probe: https://pastebin.com/BTGzYrbN

**Reply by ph0nebooth** [reply, marked as answer]
Ok, I'm an idiot. Turns out you can only have a [total of 1x backchannel per **camera**, not per **stream**](https://github.com/AlexxIT/go2rtc/issues/49#issuecomment-1828701415).

I have both a substream and mainstream configured. I disabled the backchannel on my substream, then used Nick's recommended config above. Everything is working now. Two-way audio and the built-in siren.

Final config:

    Terrace:
      - rtsp://cctv:redacted@192.168.178.202:554/cam/realmonitor?channel=1&subtype=0#backchannel=0
      - ffmpeg:Terrace#audio=opus
      - rtsp://cctv:redacted@192.168.178.202:554/cam/realmonitor?channel=1&subtype=0
    Terrace_sub:
      - rtsp://cctv:redacted@192.168.178.202:554/cam/realmonitor?channel=1&subtype=2#backchannel=0
      - ffmpeg:Terrace_sub#audio=opus

@NickM-27 I also added the recommended ffmpeg presets according to the Frigate documentation. Can you please confirm if they're necessary in my scenario?

    Terrace: # <------ Name the camera
      enabled: true
      ffmpeg:
        output_args:
          record: preset-record-generic-audio-copy
        inputs:
          - path: rtsp://127.0.0.1:8554/Terrace # <----- The stream you want to use for detection
            input_args: preset-rtsp-restream
            roles:
              - record
          - path: rtsp://127.0.0.1:8554/Terrace_sub # <----- The stream you want to use for detection
            roles:
              - detect

Grateful for all your help.
